server/game/cards/02_SHD/units/TarffulKashyyykChieftain.ts

```diff
@@ -0,0 +1,36 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { DamageType, RelativePlayer, Trait, WildcardCardType, ZoneName } from '../../../core/Constants';
+
+export default class TarffulKashyyykChieftain extends NonLeaderUnitCard {
+    protected override getImplementationId () {
+        return {
+            id: '7022736145',
+            internalName: 'tarfful#kashyyyk-chieftain'
+        };
+    }
+
+    public override setupCardAbilities () {
+        this.addTriggeredAbility({
+            title: 'That unit deals that much damage to an enemy ground unit',
+            when: {
+                onDamageDealt: (event, context) =>
+                    event.type === DamageType.Combat &&
+                    event.card.hasSomeTrait(Trait.Wookiee) &&
+                    event.isResolved &&
```

Comment:
We shouldn't check `event.isResolved` explicitly like this, it could cause issues. We fire the ability trigger before and after the event resolves partly because of the need to catch triggers on units that are leaving the field during resolution.

An example where this wouldn't work is if the new Maul unit attacks both Tarfful and another wookiee at the same time, and defeats Tarfful but not the other wookiee. Tarfful's ability would not fire afterwards, because he was defeated during the attack and his triggers unregistered immediately after. 

As for the way to actually check this condition, I think it may actually be enough to just check if the event's damage is enough to defeat the unit. There can't be multiple damage events to the same target in the same window due to rule 1.9.8 (linked below), so I'm pretty sure we can safely do that check and assume it will be correct. Lmk if you can think of an exception though.

![image](https://github.com/user-attachments/assets/28e30f63-e5aa-4413-a3e0-adee7b4c9901)

There is one hiccup there which is we need to know if the damage has already been applied or not when we check the unit's `remainingHp` property, because it might be the case that we're seeing the event after it's already resolved. Probably the safest thing to do is to add a new property on the event itself called `willDefeat` or something and just check that.